server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -1174,15 +1174,14 @@
         Pair<List<DomainRouterJoinVO>, Integer> result = searchForRoutersInternal(cmd, cmd.getId(), cmd.getRouterName(), cmd.getState(), cmd.getZoneId(), cmd.getPodId(), cmd.getClusterId(),
                 cmd.getHostId(), cmd.getKeyword(), cmd.getNetworkId(), cmd.getVpcId(), cmd.getForVpc(), cmd.getRole(), cmd.getVersion());
         ListResponse<DomainRouterResponse> response = new ListResponse<DomainRouterResponse>();
+        if (!VirtualNetworkApplianceManager.RouterHealthChecksEnabled.value()) {
+            s_logger.debug("Skipping health checks data in router as router.health.checks.enabled is false for router");
+            return response;
+        }
```

Comment:
again, i am more interested in seeing a log when it is enabled.

Reply:
Daan this is a debug log and returned as per API. The extended logics are logged in method calls. thoughtS?